server/status.go

```diff
@@ -39,68 +38,73 @@ import (
-	// nodes that aren't communicating with the cluster properly.
-	statusLocalKeyPrefix = statusKeyPrefix + "local/"
-
-	// statusLocalLogKeyPrefix exposes a list of log files for the node.
-	// logfiles -> lists available log files
-	// logfiles/ -> lists available log files
-	// logfiles/{file} -> fetches contents of named log
-	statusLocalLogFileKeyPrefix = statusLocalKeyPrefix + "logfiles/"
-	// statusLocalLogFileKeyPattern is the pattern to match
-	// logfiles/{file}
-	statusLocalLogFileKeyPattern = statusLocalLogFileKeyPrefix + ":file"
-
-	// statusLocalStacksKey exposes stack traces of running goroutines.
-	statusLocalStacksKey = statusLocalKeyPrefix + "stacks"
-
-	// statusNodeKeyPrefix exposes status for each of the nodes the cluster.
-	// nodes -> lists all nodes
-	// nodes/ -> lists all nodes
-	// nodes/{NodeID} -> shows only the status for that specific node
-	statusNodeKeyPrefix = statusKeyPrefix + "nodes/"
-	// statusNodeKeyPattern is the pattern to match nodes/{NodeID}
-	statusNodeKeyPattern = statusNodeKeyPrefix + ":id"
-
-	// statusStoreKeyPrefix exposes status for each store.
-	// stores -> lists all nodes
-	// stores/ -> lists all nodes
-	// stores/{StoreID} -> shows only the status for that specific store
-	statusStoreKeyPrefix = statusKeyPrefix + "stores/"
-	// statusStoreKeyPattern is the pattern to match stores/{StoreID}
-	statusStoreKeyPattern = statusStoreKeyPrefix + ":id"
-
-	// statusLogKeyPrefix exposes the logs for each node
-	// logs -> list log entries for the local node
-	// logs/ -> list log entries for the local node
-	// logs/{NodeID} -> list logs entries for that specific node
-	statusLogKeyPrefix = statusKeyPrefix + "logs/"
-	// statusLogKeyPattern is the pattern to match log/{NodeID}
-	statusLogKeyPattern = statusLogKeyPrefix + ":node_id"
-
-	// statusTransactionsKeyPrefix exposes transaction statistics.
-	statusTransactionsKeyPrefix = statusKeyPrefix + "txns/"
-
+	/*
+		Note that :node_id can always be replaced by the value "local" to see
+		the local nodes response.
+
+		/_status/details/:node_id		 - specific node's details
+		/_status/gossip/:node_id         - specific node's gossip
+		/_status/logfiles/:node_id       - list log files
+		/_status/logfiles/:node_id/:file - returns the contents of the specific
+										   log files on specific node
+		/_status/logs/:node_id           - log entries from a specific node
+		/_status/stacks/:node_id		 - exposes stack traces of running
+										   goroutines
+		/_status/nodes				     - all nodes' status
+		/_status/nodes/:node_id		     - a specific node's status
+		/_status/stores                  - all stores' status
+		/_status/stores/:store_id        - a specific store's status
+	*/
+
+	// statusPrefix is the root of the cluster statistics and metrics API.
+	statusPrefix = "/_status/"
+
+	// statusGossipPattern exposes a view of the gossip network.
+	statusGossipPattern = "/_status/gossip/:node_id"
```

Comment:
I find this version with the "duplication" a lot more readable than the one with the constants. There's no compile-time verification that `/_status/gossip/` and `/_status/details/` start with the same string, but so what? Whenever I read the old version of this code that assembled url strings from pieces I was afraid we'd end up with missing (or doubled) slashes.